metadata/metadata.go

```diff
@@ -1690,107 +1792,171 @@ func (serv *MetadataServer) CreateSourceVariant(ctx context.Context, variant *pb
+	logger.Info("Creating Model")
+	return serv.genericCreate(ctx, &modelResource{modelRequest.Model}, nil)
 }
 
 func (serv *MetadataServer) GetModels(stream pb.Metadata_GetModelsServer) error {
-	return serv.genericGet(stream, MODEL, func(msg proto.Message) error {
+	ctx := logging.AddLoggerToContext(stream.Context(), serv.Logger)
+	serv.Logger.Info("Opened Get Models stream")
+	return serv.genericGet(ctx, stream, MODEL, func(msg proto.Message) error {
 		return stream.Send(msg.(*pb.Model))
 	})
 }
 
 // GetEquivalent attempts to find an equivalent resource based on the provided ResourceVariant.
-func (serv *MetadataServer) GetEquivalent(ctx context.Context, req *pb.ResourceVariant) (*pb.ResourceVariant, error) {
-	return serv.getEquivalent(req, true)
+func (serv *MetadataServer) GetEquivalent(ctx context.Context, req *pb.ResourceVariantRequest) (*pb.ResourceVariant, error) {
+	_, ctx, logger := serv.Logger.InitializeRequestID(ctx)
+	logger.Info("Getting Equivalent Resource Variant, %v", req.ResourceVariant.Resource)
+	return serv.getEquivalent(ctx, req.ResourceVariant, true)
 }
 
 /*
 *
 This method is used to get the equivalent resource variant for a given resource variant. readyStatus is used to determine
 if we should only return the equivalent resource variant if it is ready.
 */
-func (serv *MetadataServer) getEquivalent(req *pb.ResourceVariant, filterReadyStatus bool) (*pb.ResourceVariant, error) {
+func (serv *MetadataServer) getEquivalent(ctx context.Context, req *pb.ResourceVariant, filterReadyStatus bool) (*pb.ResourceVariant, error) {
 	noEquivalentResponse := &pb.ResourceVariant{}
-
+	logger := logging.GetLoggerFromContext(ctx)
 	currentResource, resourceType, err := serv.extractResourceVariant(req)
 	if err != nil {
+		logger.Errorw("Error extracting resource variant", "resource variant", req, "error", err)
 		return nil, err
 	}
+	resourceProto := currentResource.ToResourceVariantProto()
+	var resourceName string
+	var resourceVariant string
+	var loggingResourceType logging.ResourceType
+	switch resourceType {
+	case SOURCE_VARIANT:
+		resourceName = resourceProto.GetSourceVariant().Name
+		resourceVariant = resourceProto.GetSourceVariant().Variant
+		loggingResourceType = logging.SourceVariant
+	case FEATURE_VARIANT:
+		resourceName = resourceProto.GetFeatureVariant().Name
+		resourceVariant = resourceProto.GetFeatureVariant().Variant
+		loggingResourceType = logging.FeatureVariant
+	case LABEL_VARIANT:
+		resourceName = resourceProto.GetLabelVariant().Name
+		resourceVariant = resourceProto.GetLabelVariant().Variant
+		loggingResourceType = logging.LabelVariant
+	case TRAINING_SET_VARIANT:
+		resourceName = resourceProto.GetTrainingSetVariant().Name
+		resourceVariant = resourceProto.GetTrainingSetVariant().Variant
+		loggingResourceType = logging.TrainingSetVariant
+	default:
+		return nil, fferr.NewInvalidArgumentError(fmt.Errorf("unknown resource variant type: %T", req.Resource))
```

Comment:
Since this is only being used for logging, I'd prefer just printing a warning and let things ride